openfga/v1/openfga_service.proto

```diff
@@ -1173,6 +1196,59 @@ message CheckResponse {
   string resolution = 2;
 }
 
+message BatchCheckRequest {
+  string store_id = 1 [
+    json_name = "store_id",
+    (google.api.field_behavior) = REQUIRED,
+    (validate.rules).string = {pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01YCP46JKYM8FJCQ37NMBYHE5X\""}
+  ];
+
+  repeated BatchCheckItem checks = 2 [(google.api.field_behavior) = REQUIRED];
+
+  openfga.v1.ContextualTupleKeys contextual_tuples = 3 [json_name = "contextual_tuples"];
+
+  google.protobuf.Struct context = 4;
+
+  string authorization_model_id = 5 [
+    json_name = "authorization_model_id",
+    (validate.rules).string = {
+      pattern: "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$"
+      ignore_empty: true
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {example: "\"01G5JAVJ41T49E9TT3SKVS7X1J\""}
+  ];
+
+  ConsistencyPreference consistency = 6 [(validate.rules).enum.defined_only = true];
```

Comment:
It seems arbitrary that you are allowed to specify per-request context and contextual tuples, but not model ID or consistency.

Reply:
Was the result that came out of [this RFC](https://oktawiki.atlassian.net/wiki/spaces/SDCSTL/pages/3097502917/RFC+Batch+Check+Request+Response+Interface). But since I opened this PR that RFC has gotten new traction so this PR is likely to change

Reply:
converted back to draft while the RFC is ironed out further